numpy/f2py/crackfortran.py

```diff
@@ -2536,10 +2559,10 @@ def get_parameters(vars, global_params={}):
                     l = markoutercomma(v[1:-1]).split('@,@')
 
             try:
-                params[n] = eval(v, g_params, params)
+                params[n] = param_eval(n, vars[n], v, g_params, params)
             except Exception as msg:
-                params[n] = v
-                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(n)))
+
             if isstring(vars[n]) and isinstance(params[n], int):
```

Comment:
Won't this line now raise `KeyError` because `params[n]` is no longer set by the `except` block?

Reply:
Yes. Following the suggestion from @pearu above I'm not raising another exception, but I've added the unparsed parameter array as `params[n]` so this doesn't cause trouble further down.